frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -675,465 +919,376 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
     return null;
   };
 
-  const getGroupName = (group, groupType) => {
-    if (!_.isString(group) || !_.isString(groupType)) {
-      return null;
-    }
-    let groupName = group.split(groupType)[1];
-    if (groupType.startsWith(SpecCapability.arrayFieldGroup)) {
-      [groupName] = groupName.split(':');
-    }
-    return groupName;
-  };
+  /* Inserts an arrayFieldGroup at nextIndex. The fieldList argument is an array of fields
+   * representing a single element in an arrayFieldGroup. To add an arrayFieldGroup, we duplicate
+   * each field in this fieldList, replacing the array index in field.path with the nextIndex
+   * argument
+   */
+  const addArrayFieldGroup = (fieldLists: OperandField[][]) => {
```

Comment:
Replaces previous `addFieldsToGroup`, `addGroup`, and `addGroupAndFields` helper functions. Unified into a single helper.